src/nspawn/nspawn-seccomp.c

```diff
@@ -29,111 +29,155 @@
 #include "alloc-util.h"
 #include "log.h"
 #include "nspawn-seccomp.h"
+
 #ifdef HAVE_SECCOMP
 #include "seccomp-util.h"
-#endif
 #include "string-util.h"
+#include "strv.h"
+#endif
 
 #ifdef HAVE_SECCOMP
 
-static int seccomp_add_default_syscall_filter(
-                scmp_filter_ctx ctx,
-                uint32_t arch,
-                uint64_t cap_list_retain) {
-
-        static const struct {
-                uint64_t capability;
-                int syscall_num;
-        } blacklist[] = {
-                { 0,              SCMP_SYS(_sysctl)             }, /* obsolete syscall */
-                { 0,              SCMP_SYS(add_key)             }, /* keyring is not namespaced */
-                { 0,              SCMP_SYS(afs_syscall)         }, /* obsolete syscall */
-                { 0,              SCMP_SYS(bdflush)             },
+static const SyscallFilterSet nspawn_default_blacklist = {
```

Comment:
I would move this into here 
https://github.com/kinvolk/systemd/blob/master/src/shared/seccomp-util.c#L270
and document it here 
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#SystemCallFilter=

This would make it much easier to find the reason for the bug we had originally.
Going to ask that in the PR.